### PR TITLE
Add `bump-crate` script

### PR DESCRIPTION
**Pull request by buffalojoec**
#### Problem
Each time we need to increment the version of a crate within SPL - either major, minor, or patch - we're doing it manually. It would be nice to have this action scripted.

#### Summary of Changes
Here's a script!

### CODE REVIEW

**Review by joncinque — COMMENTED** [review]
Just a couple of comments -- looks great otherwise!

File: bump-crate.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
```
Comment (joncinque) [inline bump-crate.sh:1]: Oh and last one, typically it's better to declare these in a totally portable way
```suggestion
#!/usr/bin/env bash
```

File: bump-crate.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  patch)
+    patch=$((patch + 1))
+    ;;
+esac
+
+new_version="$major.$minor.$patch"
+echo "New version: $new_version"
+
+# Increment version in each manifest of the specified package.
+sed -i '' -e "s/^version.*$/version = \"$new_version\"/g" "$manifest_path"
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply bump-crate.sh:58]: Unfortunately I think this breaks on Mac. When I run it without the space I get a bunch of `Cargo.toml-e` files. Haven't dug too deep, though, but if it's not working on Linux I can maybe try to find some middle ground.

[the remark this answers is not in the document]
Comment (joncinque) [reply bump-crate.sh:58]: Ah right, I always forget about that... does it work on mac without the `''`? as in `-i` on its own?

[the remark this answers is not in the document]
Comment (buffalojoec) [reply bump-crate.sh:58]: We could maybe do something like this, but it ain't pretty.

```bash
if [[ "$OSTYPE" == "darwin"* ]]; then
    SED_CMD="sed -i ''"
else
    SED_CMD="sed -i"
fi

$SED_CMD -e "s/^version.*$/version = \"$new_version\"/g" "$manifest_path"
```

**Pull request closed without merging**